src/core/Quote.ml

```diff
@@ -54,21 +54,37 @@ let rec quote_con (tp : D.tp) con =
   | D.TpPrf _, _ ->
     ret S.Prf
   | _, D.Split branches ->
-    let quote_branch (phi, clo) =
-      lift_cmp @@ CmpM.test_sequent [phi] CofBuilder.bot |>> function
-      | false ->
-        let+ tphi = quote_cof phi
-        and+ tbdy =
-          bind_var (D.TpPrf phi) @@ fun prf ->
-          let* body = lift_cmp @@ inst_tm_clo clo prf in
-          quote_con tp body
-        in
-        Some (tphi, tbdy)
-      | true ->
-        ret None
+    let rec reduce_if_true = function
```

Comment:
This seems like a useful little function! Perhaps we can factor it out to `Semantics.ml`?

Reply:
We actually already had it in `Semantics.ml`, it just had to be given a name haha